src/decomp_2d.f90

```diff
@@ -406,10 +399,18 @@ subroutine decomp_info_init(nx, ny, nz, decomp)
       buf_size = max(decomp%xsz(1) * decomp%xsz(2) * decomp%xsz(3), &
                      max(decomp%ysz(1) * decomp%ysz(2) * decomp%ysz(3), &
                          decomp%zsz(1) * decomp%zsz(2) * decomp%zsz(3)))
+
 #ifdef EVEN
       ! padded alltoall optimisation may need larger buffer space
       buf_size = max(buf_size, &
                      max(decomp%x1count * dims(1), decomp%y2count * dims(2)))
+      ! evenly distributed data ?
+      if (mod(nx, dims(1)) == 0 .and. mod(ny, dims(1)) == 0 .and. &
+          mod(ny, dims(2)) == 0 .and. mod(nz, dims(2)) == 0) then
+         decomp%even = .true.
+      else
+         decomp%even = .false.
+      end if
 #endif
```

Comment:
I would add a 
```
#else 
  decomp%even = .false.
#endif
```
to ensure there is a default value in case it is tested elsewhere.

Reply:
the variable does not exist if the preprocessor variable is not present

Reply:
ah, ok